prime-router/src/main/kotlin/azure/ApiSearch.kt

```diff
@@ -190,23 +231,31 @@ abstract class ApiSearch<PojoType, RecordType : Record, ApiFilterType : ApiFilte
      * @return the list of the records parsed into the [PojoType]
      *
      */
-    open fun <T : Record> fetchResults(dslContext: DSLContext, select: SelectJoinStep<T>): ApiSearchResult<PojoType> =
+
+    private fun <T : Record, R : Record> fetchResults(
+        dslContext: DSLContext,
+        selectFields: SelectFieldOrAsterisk,
```

Comment:
The problem this solves is that the JOOQ fluent API is mutable so
```
val selectWithWhere = select.where(getWhereClause())
```
will actually modify `select`. Passing in fields like this made it much easier to run them in parallel